resources/users/router.js

```diff
@@ -1,15 +1,31 @@
 const express = require('express');
 
-const { deleteUser, getUserScore, getLeaderboard } = require('./controller');
+const {
+  deleteUser,
+  getUserScore,
+  getLeaderboard,
+  updateUserProfile,
+} = require('./controller');
+
+const validate = require('../../utils/validate');
+
+const { updateUserProfileSchema } = require('./userSchema');
 
 const { authorized } = require('../global/middlewares');
 
 const userRouter = express.Router();
 
 userRouter.delete('/:id', authorized, deleteUser);
 
-userRouter.get('/:id/score', getUserScore);
+userRouter.get('/:id/score', authorized, getUserScore);
 
 userRouter.get('/leaderboard', getLeaderboard);
 
+userRouter.post(
+  '/updateprofile',
+  authorized,
```

Comment:
Can you remove the authorized middleware, The authorized middleware has been used already on the apiRouter for users route